cpp/include/ray/api/ray_config.h

```diff
@@ -16,10 +17,9 @@ class RayConfig {
   // Whether or not to run this application in a local mode. This is used for debugging.
   bool local_mode = false;
 
-  // The dynamic library path which contains remote fuctions of users.
-  // This parameter is not used when the application runs in local mode.
-  // TODO(guyang.sgy): Put this param into job config instead.
-  std::string dynamic_library_path = "";
+  // An array of directories or dynamic library files that specify the search path for
+  // user code. This parameter is not used when the application runs in local mode.
+  std::vector<std::string> code_search_path;
```

Comment:
We should create a `job_config` field and put `code_search_path` into it. We may add more job config options in the future.

Reply:
Why use two struct? Maybe one is enough?

Reply:
Please refer to `ray.default.conf` for Java job config structures. And in Python, we use ray.init(..., job_config=ray.job_config.JobConfig(code_search_path=...)). So it's already two levels in both Java and Python.

Reply:
Python don't has RayConfig, so it should make a JobConfig to prevent a big number arguments of Ray.init function?

Reply:
I don't think the two-level structure has anything to do with flattened parameters of `ray.init()` in Python. We designed the JobConfig class in Python and `ray.job.xxx` system property section in Java because we want to inform users that these options will have some impact on jobs, and they only impact jobs, not nodes or the cluster. That is, options designed to tune a job's behavior and have no impact on nodes or the cluster should be put in job config.

Reply:
From another perspective, options in job config will not be shared by jobs. They are job-specific. Options not in job config are shared by all jobs. They are usually node or cluster options.

Reply:
I think all the items in RayConfig should be job-specific.

Reply:
We will make another PR to do this. Issue https://github.com/ray-project/ray/issues/16994.